app/code/Magento/Downloadable/Observer/IsAllowedGuestCheckoutObserver.php

```diff
@@ -3,65 +3,121 @@
  * Copyright © Magento, Inc. All rights reserved.
  * See COPYING.txt for license details.
  */
+
 namespace Magento\Downloadable\Observer;
 
+use Magento\Downloadable\Model\Product\Type;
+use Magento\Downloadable\Model\ResourceModel\Link\CollectionFactory;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\Framework\Event\Observer;
 use Magento\Framework\Event\ObserverInterface;
+use Magento\Quote\Api\Data\CartItemInterface;
+use Magento\Quote\Model\Quote;
 use Magento\Store\Model\ScopeInterface;
 
+/**
+ * Checks if guest checkout is allowed then quote contains downloadable products.
+ */
 class IsAllowedGuestCheckoutObserver implements ObserverInterface
 {
     /**
      *  Xml path to disable checkout
      */
-    const XML_PATH_DISABLE_GUEST_CHECKOUT = 'catalog/downloadable/disable_guest_checkout';
+    private const XML_PATH_DISABLE_GUEST_CHECKOUT = 'catalog/downloadable/disable_guest_checkout';
+
+    /**
+     *  Xml path to get downloadable Shareable setting
+     */
+    private const XML_PATH_DOWNLOADABLE_SHAREABLE = 'catalog/downloadable/shareable';
```

Comment:
`const` name is descriptive-enough, you don't need to add docblock
```suggestion
    private const XML_PATH_DOWNLOADABLE_SHAREABLE = 'catalog/downloadable/shareable';
```